corehq/tests/pytest_plugins/reusedb.py

```diff
@@ -442,4 +442,15 @@ def unblock():
     return block, unblock
 
 
+class BlockedMock(Mock):
+    def __getattr__(self, name):
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            raise RuntimeError(
+                'Database access not allowed, use the "django_db" mark, or '
+                'the "db" or "transactional_db" fixtures to enable it.'
+            )
```

Comment:
As mentioned in https://github.com/dimagi/commcare-hq/pull/34778#discussion_r1854551526, I don't think me commenting on this originally was worthwhile. I was mainly pondering if there was a way to get a `RuntimeError` "organically", not to create a mock that translates an AttributeError into a RuntimeError just for the sake of matching the other dbs in the test. I'm open to removing this if you are.

Reply:
I do think this error message is better than the attribute error, so it does seem like an improvement. Given that the [other source of this error](https://github.com/pytest-dev/pytest-django/blob/1ffc3239387bcfadbd8aecfdfacf3d6b52b947f8/pytest_django/plugin.py#L832-L836) is `pytest-django`, it seems reasonable for this module to raise a similar error for Couch.